timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -102,14 +105,24 @@ public void createAndRegisterResources() {
         registerExceptionMappers();
         leadershipCreator.registerLeaderElectionService();
 
-        // Finally, register the endpoints associated with the clients.
+        // Finally, register the health check, and endpoints associated with the clients.
+        healthCheckSupplier = leadershipCreator.getHealthCheck();
         registrar.accept(
                 new TimeLockResource(this::createInvalidatingTimeLockServices,
                         JavaSuppliers.compose(TimeLockRuntimeConfiguration::maxNumberOfClients, runtime)));
 
         ClockSkewMonitorCreator.create(install, registrar).registerClockServices();
     }
 
+    @SuppressWarnings("unused") // used by external health checks
+    public TimeLockStatus getStatus() {
+        if (healthCheckSupplier != null) {
```

Comment:
can we make this an `Optional` rather than nullable?

Reply:
Well then we'd have a warning about having an optional field, plus `healthCheckSupplier.get().get().getStatus()`, which just seems manky.

I think it would be cleaner to instead have a create method on `TimeLockAgent` which calls `createAndRegisterResources`, so that `healthCheckSupplier` is never null by the time TimeLock gets to it, and then `getStatus` can never be called when the supplier is not there.